src/dbg/debugger.cpp

```diff
@@ -853,6 +849,94 @@ void cbTICNDStep()
     }
 }
 
+void cbTIBTStep()
+{
+    // Trace record
+    duint CIP = GetContextDataEx(hActiveThread, UE_CIP);
+    if(!traceCondition)
+    {
+        _dbg_dbgtraceexecute(CIP);
+        dprintf("Bad tracing state.\n");
+        cbRtrFinalStep();
+        return;
+    }
+    if((TraceRecord.getTraceRecordType(CIP) != TraceRecordManager::TraceRecordNone && TraceRecord.getHitCount(CIP) == 0) || !traceCondition->ContinueTrace())

```

Comment:
Oh I thought you also modified the current tracing, my bad.
On Thu, 16 Jun 2016 at 18:33, Torusrxxx <notifications@github.com> wrote:

> In src/dbg/debugger.cpp
> <https://github.com/x64dbg/x64dbg/pull/756#discussion_r67379172>:
>
> > @@ -853,6 +849,94 @@ void cbTICNDStep()
> >      }
> >  }
> >
> > +void cbTIBTStep()
> > +{
> > +    // Trace record
> > +    duint CIP = GetContextDataEx(hActiveThread, UE_CIP);
> > +    if(!traceCondition)
> > +    {
> > +        _dbg_dbgtraceexecute(CIP);
> > +        dprintf("Bad tracing state.\n");
> > +        cbRtrFinalStep();
> > +        return;
> > +    }
> > +    if((TraceRecord.getTraceRecordType(CIP) != TraceRecordManager::TraceRecordNone && TraceRecord.getHitCount(CIP) == 0) || !traceCondition->ContinueTrace())
>
> This command is trace record oriented. Condition is often set to "0".
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/x64dbg/x64dbg/pull/756/files/36c5e3d37492d08cc8af71ed7ae7cf22e4756325#r67379172>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACWCmc5Rbfsipo-KjVFiVq5STDbIIuG6ks5qMXrugaJpZM4I3kEa>
> .
>